<?xml version="1.0"?>

<launch>
  <arg name="ip_address" default="172.16.0.1"/>
  <arg name="port" default="4000"/>
  <arg name="max_linear_vel" default="0.2"/>
  <arg name="max_angular_vel" default="1"/>
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
    <param name="port" value="/dev/ttyACM0"/>
  </node>

  <node pkg="cmd_sub" type="cmd_sub_node" name="cmd_sub" respawn="false" output="screen">

    <param name="ip_address" value="$(arg ip_address)"/>
    <param name="port" value="$(arg port)"/>
    <param name="max_linear_vel" value="$(arg max_linear_vel)"/>
    <param name="max_angular_vel" value="$(arg max_angular_vel)"/>

  </node>

  <node pkg="ball_detection" type="ball_detect_node2" name="ball_detect2"/>
  <node pkg="ball_detection" type="modify_ball_count_node2" name="modify_ball_count2"/>
  <node pkg="data_integrate" type="ball_collect_node" name="ball_collect"/>
  <node pkg="cmd_sub" type="ball_count_node" name="ball_count"/>


</launch>
